### PR TITLE
Update Github Badge for Workflow

### PR DESCRIPTION
**Pull request by huayuan4396**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by huayuan4396** [comment]
May have to enable the Read and Write Permissions in the repo workflow setting.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -316,3 +316,101 @@ jobs:
           conda activate $pwd\tmp_env
           $env:path = "$pwd\build\bin\Release;" + $env:path
           .github\scripts\windows\test_full_pipeline.ps1 -Backend trt -Device cuda
+
+  badge_build_cpu_model_convert:
```
Comment (RunningLeon) [inline .github/workflows/build.yml:320]: @huayuan4396  could we put this step and the end of job `build_cpu_model_convert` as last step?

**Comment by RunningLeon** [comment]
@huayuan4396 it seems the `RubbaBoy/BYOB@v1.2.1` only works on linux

**Comment by huayuan4396** [comment]
> @huayuan4396 it seems the `RubbaBoy/BYOB@v1.2.1` only works on linux

May have to do this by using a job instead of step for windows.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -251,10 +301,20 @@ jobs:
           export LD_LIBRARY_PATH="/root/workspace/mmdeploy/build/lib:${LD_LIBRARY_PATH}"
           python3 tools/check_env.py
       - name: Test TensorRT pipeline
+        id: badge_status
```
Comment (RunningLeon) [inline .github/workflows/build.yml:304]: why add id here?

Comment (huayuan4396) [reply .github/workflows/build.yml:304]: > why add id here?
Get the status of the step before badge step to know if previous steps are all success or not.

**Comment by RunningLeon** [comment]
@huayuan4396 Could sync with latest `main` branch to fix all ci errors.

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**